A public roadmap tool, based on Zendesk community (#3)
zendesk/demo-snippets: community-roadmap/README.md

> # Community Roadmap for Zendesk Help Center

Reviewer: ```suggestion
# Community roadmap for a Zendesk help center
```

> This project provides a public roadmap feature in Zendesk Help Center, where the product roadmap items are driven by community posts.

Reviewer: ```suggestion
This project provides a public roadmap feature in Zendesk Guide. The product roadmap items are driven by community posts.
```

> 

Reviewer: It's up to you, but I think it may be more helpful to provide more context here:

- What problem does this solve? Why would a user be interested in doing this? Does it let them identify roadmap items based on community interest?
- At a high-level, how do community posts determine the roadmap items?
- Are there some example screenshots or a demo link we can share?
- What prerequisites or skills are needed to set up the feature? Do users need a specific plan? Do they need experience with React or node.js?

Author: Good feedback, thx. I've updated the READMEs, please take a look.

## Structure

The project is divided into two major modules:

The `backend` module is a REST API service built for hosting outside of Zendesk.

> The `frontend` module is the  client of the REST API, built to run inside Zendesk Help Center, for instance as a Custom Page.

Reviewer: Unless I'm mistaken, there isn't any other way for a user to embed an app into a help center.
```suggestion
The `frontend` module is the  client of the REST API, built to run inside a help center as a custom page.
```

# Community Roadmap for Zendesk Guide

This project provides a public roadmap feature in Zendesk Guide, where the product roadmap items are driven by community posts.

> The idea is that customers leave feedback and feature requests in the community. This tool makes it possible to create a publically visible roadmap where each item in the roadmap has a linked community post. Users can vote and comment on the roadmap items simply by voting and commenting in the community. As such it is a specialized and curated view into the community.

Reviewer: ```suggestion
Customers can leave feedback and feature requests in a Gather community. The tool creates a publicly visible roadmap. Each item in the roadmap is linked to a Gather community post. Users can vote and comment on the roadmap items by voting and commenting on the linked community post.
```

![Screenshot of a roadmap created with this tool](screenshot.png "Screenshot")

## Structure

The project defines this structure of a roadmap:

* Product: A roadmap is created for a single product. A product has a set of owners who are the ones who can modify the roadmap.
* Product area: An area of the product that you wish to place roadmap items into.
* Product area item: An individual feature or project that you are considering or planning to do.

> The service is built to host multiple products, and thus multiple roadmaps. These are kept fully independent from each other, but you don't need multiple service instances just because you have multiple products.

Reviewer: ```suggestion
The service is built to host multiple products, producing a roadmap for each product. However, each roadmap is fully independent. You don't need multiple service instances for multiple products.
```

## Using the service

Once the service is built and deployed, you simply use it by adding an element to your help center theme's HTML code, like this:

```
<div class="container">
	<div class="roadmapContainer" data-product-key="{my_product_key}"></div>
</div>
<script src="{frontend_script_src}"></script>
```

Replace the following placeholder strings:

* `{my_product_key}`: A unique key for your roadmapped product. For example `guide`
* `{frontend_script_src}`: The URL to the frontend script that you've built and deployed. Typically ending with `/js/main.(version).js`.

There is an example roadmap deployed on this URL: https://z3n-kaspersor.zendesk.com/hc/p/roadmap

## Codebase overview

The project is divided into two major modules:

The [backend](backend/README.md) module is a REST API service built for hosting outside of Zendesk.

> The [frontend](frontend/README.md) module is the client of the REST API, built to run inside Zendesk help center, for instance as a custom page.

Reviewer: ```suggestion
The [frontend](frontend/README.md) module is the client of the REST API, built to run inside a Zendesk help center's custom page.
```
